### PR TITLE
OAK-11171:  Added cached external membership benchmarks

### PR DESCRIPTION
**Pull request by Amoratinos**
## Summary

Created new benchmark to test external membership login with external groups and user added to local groups to compare the performance using cache.

### CODE REVIEW

**Review by anchela — COMMENTED** [review]
hi @Amoratinos , can you limit this PR to changes to benchmarks only? otherwise i would rather merge it into the general fix PR. wdyt?

File: oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/PrincipalMembershipReaderImpl.java
Comment (anchela) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/PrincipalMembershipReaderImpl.java:1]: this class should not be part of the benchmark PR, should it?

File: oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/principal/PrincipalMembershipReaderFactory.java
Comment (anchela) [inline oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/principal/PrincipalMembershipReaderFactory.java:1]: also the interface in security-spi should not be needed for adding/changing benchmarks. is suspect that's just a left-over from some other refactoring. no?

**Review by anchela — CHANGES_REQUESTED** [review]
hi @Amoratinos , there are 2 new classes (one in oak-core and one in security-spi) that imho should not be part of this PR, which just adds/modifies benchmarks. i suspect that's only a left over from some other refactoring. can you check?

File: oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/PrincipalMembershipReaderImpl.java
Comment (Amoratinos) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/PrincipalMembershipReaderImpl.java:1]: Yes, as I rebased on trunk this was a left over

File: oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/principal/PrincipalMembershipReaderFactory.java
Comment (Amoratinos) [reply oak-security-spi/src/main/java/org/apache/jackrabbit/oak/spi/security/principal/PrincipalMembershipReaderFactory.java:1]: Same here, as you mentioned a left-over after the rebase

**Review by anchela — APPROVED** [review]
lgtm! thanks a lot.

**Pull request merged**